platform/api/oci/instance.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oci
+
+import (
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/oracle/bmcs-go-sdk"
+
+	"github.com/coreos/mantle/platform"
+)
+
+func (a *API) CreateInstance(name, userdata string) (*Machine, error) {
```

Comment:
`name` isn't actually used.  It should probably go in both the `hostnameLabel` and the instance display name.